File: src/content/ruby/factor.rb
print "Enter a number: "
number = gets.chomp.to_i

factors = []
divisor = 2

while number > 1
    if number % divisor == 0
        factors << divisor
        number /= divisor
    else
        divisor += 1
    end
end

puts "Prime factors: #{factors.join(', ')}"

File: src/content/js/factor.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function getPrimeFactors(num) {
    let factors = [];
    let divisor = 2;

    while (num > 2) {
        if (num % divisor === 0) {
            factors.push(divisor);
            num = num / divisor;
        } else {
            divisor++;
        }
    }

    return factors;
}

rl.question('Enter a number: ', (num) => {
    const primeFactors = getPrimeFactors(parseInt(num));
    console.log('Prime factors:', primeFactors);
    rl.close();
});